File: 프로그래머스/unrated/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    col -= 1; row_begin -= 1; row_end -= 1
    R = len(data); C = len(data[0])
    
    data.sort(key=lambda x: x[0], reverse=True)
    data.sort(key=lambda x: x[col])
    
    hs = 0
    for r in range(row_begin, row_end+1):
        tmp = 0
        for c in range(C):
            tmp += data[r][c] % (r+1)
        hs = hs ^ tmp
    return hs

File: 프로그래머스/unrated/147354. 테이블 해시 함수/테이블 해시 함수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int column;

bool cmp(const vector<int>& v1, const vector<int>& v2){
    if (v1[column] == v2[column]){
        return v1[0] > v2[0] ? true : false;
    }
    else{
        return v1[column] < v2[column] ? true: false;
    }
}

int solution(vector<vector<int>> data, int col, int row_begin, int row_end) {
    column = col - 1;
    sort(data.begin(), data.end(), cmp);
    
    int answer = 0;
    int col_len = data[0].size();
    for (int i = row_begin; i <= row_end; i++){
        int totalSum = 0;
        for (int j = 0; j < col_len; j++){
            totalSum += (data[i-1][j] % i);
        }
        answer ^= totalSum;
    }
    return answer;
}